### DISCUSSION TITLE
High RAM consumption issue

### DISCUSSION THREAD

**Discussion by vnikoofard** [MilesCranmer/PySR#61] [Q&A] [answered]
Hi,

I would like to know is there is any way to reduce the memory consumption of the computation. My dataset is very small, 57 rows x 7 features. When I set `population=80`, `ninteration=20` the algorithm runs normally on Colab but after running for 2-3 hours (at 30% of the process) the RAM consumption increases hugely and cause the process stops (doesn't restart the kernel).  Is there any way to overcome this issue?

**Comment by MilesCranmer** [comment, marked as answer]
Hi Vahid,

Thanks for trying PySR out. Colab is built for GPU-accelerated compute such as deep learning, but PySR itself only uses CPUs (hard to do genetic algorithms on GPU), so it's not optimal. I would recommend running it on your own machine, or on a cluster with many cores if you have access to one. I would expect 80 populations or so to be too much memory usage for Colab (they have very limited RAM), so this is not unexpected behaviour. You could probably run it on Colab with fewer populations, however. Also - you should set `procs=2` (default is 4), since Colab only has 2 cores if I remember correctly.

Cheers,
Miles
